### PR TITLE
Removing Deprecated LogWrapper API

### PR DESCRIPTION
**Pull request by hiranya911**
Removing the deprecated `com.google.firebase.database.logging` package, and using SLF4J as the primary logging facade.

Some sample debug logs obtained with SLF4J and  `java.util.logging`:

```
2018-03-20 11:53:50 FINE com.google.firebase.FirebaseApp$TokenRefresher start Starting the proactive token refresher
2018-03-20 11:53:50 FINE com.google.firebase.FirebaseApp$TokenRefresher scheduleNext Scheduling next token refresh in 0 milliseconds
2018-03-20 11:53:50 FINE com.google.firebase.FirebaseApp$TokenRefresher$1 call Refreshing OAuth2 credential
2018-03-20 11:53:50 FINE com.google.firebase.FirebaseApp$TokenRefresher scheduleNext Scheduling next token refresh in 3299999 milliseconds
2018-03-20 11:53:50 FINE com.google.firebase.database.connection.PersistentConnectionImpl tryScheduleReconnect [pc_0] Scheduling connection attempt
2018-03-20 11:53:50 FINE com.google.firebase.database.connection.util.RetryHelper retry Scheduling retry in 0ms
2018-03-20 11:53:50 FINE com.google.firebase.database.connection.PersistentConnectionImpl listen [pc_0] Listening on foo (params: {})
2018-03-20 11:53:50 FINE com.google.firebase.database.connection.PersistentConnectionImpl listen [pc_0] Adding listen query: foo (params: {})
2018-03-20 11:53:50 FINE com.google.firebase.database.connection.PersistentConnectionImpl$1 run [pc_0] Trying to fetch auth token
2018-03-20 11:53:50 FINE com.google.firebase.database.connection.PersistentConnectionImpl$1$1 onSuccess [pc_0] Successfully fetched token, opening connection
2018-03-20 11:53:50 FINE com.google.firebase.database.connection.Connection open [conn_0] Opening a connection
2018-03-20 11:53:51 FINE com.google.firebase.database.core.Repo setValue set: /foo 1521572031090
2018-03-20 11:53:51 FINE com.google.firebase.database.core.view.EventRaiser raiseEvents Raising 1 event(s)
2018-03-20 11:53:51 FINE com.google.firebase.database.core.view.EventRaiser$1 run Raising /foo: VALUE: 1521572031090
2018-03-20 11:53:51 FINE com.google.firebase.database.core.Repo abortTransactions Aborting transactions for path: /foo. Affected: /foo
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection$WSClientHandlerImpl onOpen [ws_0] Websocket opened
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection resetKeepAlive [ws_0] Reset keepAlive
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection$WSClientHandlerImpl onMessage [ws_0] WS message: {"t":"c","d":{"t":"h","d":{"ts":1521572031078,"v":"5","h":"s-usc1c-nss-256.firebaseio.com","s":"99R2JhKnw6n0sTpih6zqDO0NuGe3imhv"}}}
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection resetKeepAlive [ws_0] Reset keepAlive. Remaining: 44998
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection handleNewFrameCount [ws_0] Handle new frame count: 1
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection appendFrame [ws_0] Parsed complete frame: {t=c, d={t=h, d={h=s-usc1c-nss-256.firebaseio.com, s=99R2JhKnw6n0sTpih6zqDO0NuGe3imhv, v=5, ts=1521572031078}}}
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.Connection onControlMessage [conn_0] Got control message: {t=h, d={h=s-usc1c-nss-256.firebaseio.com, s=99R2JhKnw6n0sTpih6zqDO0NuGe3imhv, v=5, ts=1521572031078}}
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.Connection onConnectionReady [conn_0] Realtime connection established
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.PersistentConnectionImpl onReady [pc_0] onReady
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.PersistentConnectionImpl handleTimestamp [pc_0] Handling timestamp
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.PersistentConnectionImpl sendConnectStats [pc_0] Sending first connection stats
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.Connection sendData [conn_0] Sending data: {t=d, d={a=s, r=0, b={c={sdk.admin_java.5-9-1-SNAPSHOT=1}}}}
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection resetKeepAlive [ws_0] Reset keepAlive. Remaining: 44996
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.PersistentConnectionImpl restoreAuth [pc_0] Calling restore state
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.PersistentConnectionImpl restoreAuth [pc_0] Restoring auth.
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.Connection sendData [conn_0] Sending data (contents hidden)
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection resetKeepAlive [ws_0] Reset keepAlive. Remaining: 44997
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection$WSClientHandlerImpl onMessage [ws_0] WS message: {"t":"d","d":{"r":0,"b":{"s":"ok","d":""}}}
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection resetKeepAlive [ws_0] Reset keepAlive. Remaining: 44946
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection handleNewFrameCount [ws_0] Handle new frame count: 1
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.WebsocketConnection appendFrame [ws_0] Parsed complete frame: {t=d, d={r=0, b={d=, s=ok}}}
2018-03-20 11:53:51 FINE com.google.firebase.database.connection.Connection onDataMessage [conn_0] Received data message: {r=0, b={d=, s=ok}}
2018-03-20 11:53:51 FINE
```

Same with the SLF4J simple logger:

```
[main] DEBUG com.google.firebase.FirebaseApp - Starting the proactive token refresher
[main] DEBUG com.google.firebase.FirebaseApp - Scheduling next token refresh in 0 milliseconds
[firebase-scheduled-worker] DEBUG com.google.firebase.FirebaseApp - Refreshing OAuth2 credential
[firebase-scheduled-worker] DEBUG com.google.firebase.FirebaseApp - Scheduling next token refresh in 3299999 milliseconds
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] Scheduling connection attempt
[firebase-database-worker] DEBUG com.google.firebase.database.connection.util.RetryHelper - Scheduling retry in 0ms
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] Listening on foo (params: {})
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] Adding listen query: foo (params: {})
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] Trying to fetch auth token
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] Successfully fetched token, opening connection
[firebase-database-worker] DEBUG com.google.firebase.database.core.Repo - set: /foo 1521572606467
[firebase-database-worker] DEBUG com.google.firebase.database.core.view.EventRaiser - Raising 1 event(s)
[firebase-database-event-target] DEBUG com.google.firebase.database.core.view.EventRaiser - Raising /foo: VALUE: 1521572606467
[firebase-database-worker] DEBUG com.google.firebase.database.core.Repo - Aborting transactions for path: /foo. Affected: /foo
[firebase-websocket-worker] DEBUG io.netty.handler.ssl.SslHandler - [id: 0x747d73bb, L:/172.18.44.82:32840 - R:admin-java-integration.firebaseio.com/35.201.97.85:443] HANDSHAKEN: TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
[firebase-websocket-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Websocket opened
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Reset keepAlive
[firebase-websocket-worker] DEBUG io.netty.handler.codec.http.websocketx.WebSocket08FrameDecoder - Decoding WebSocket Frame opCode=1
[firebase-websocket-worker] DEBUG io.netty.handler.codec.http.websocketx.WebSocket08FrameDecoder - Decoding WebSocket Frame length=132
[firebase-websocket-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] WS message: {"t":"c","d":{"t":"h","d":{"ts":1521572606524,"v":"5","h":"s-usc1c-nss-256.firebaseio.com","s":"7p9ho06FGNMsdbF5ywyoAGlqAhNW20lw"}}}
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Reset keepAlive. Remaining: 44997
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Handle new frame count: 1
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Parsed complete frame: {t=c, d={t=h, d={h=s-usc1c-nss-256.firebaseio.com, s=7p9ho06FGNMsdbF5ywyoAGlqAhNW20lw, v=5, ts=1521572606524}}}
[firebase-database-worker] DEBUG com.google.firebase.database.connection.Connection - [conn_0] Got control message: {t=h, d={h=s-usc1c-nss-256.firebaseio.com, s=7p9ho06FGNMsdbF5ywyoAGlqAhNW20lw, v=5, ts=1521572606524}}
[firebase-database-worker] DEBUG com.google.firebase.database.connection.Connection - [conn_0] Realtime connection established
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] onReady
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] Handling timestamp
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] Sending first connection stats
[firebase-database-worker] DEBUG com.google.firebase.database.connection.Connection - [conn_0] Sending data: {t=d, d={a=s, r=0, b={c={sdk.admin_java.5-9-1-SNAPSHOT=1}}}}
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Reset keepAlive. Remaining: 44996
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] Calling restore state
[firebase-database-worker] DEBUG com.google.firebase.database.connection.PersistentConnection - [pc_0] Restoring auth.
[firebase-websocket-worker] DEBUG io.netty.handler.codec.http.websocketx.WebSocket08FrameEncoder - Encoding WebSocket Frame opCode=1 length=75
[firebase-database-worker] DEBUG com.google.firebase.database.connection.Connection - [conn_0] Sending data (contents hidden)
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Reset keepAlive. Remaining: 44996
[firebase-websocket-worker] DEBUG io.netty.handler.codec.http.websocketx.WebSocket08FrameEncoder - Encoding WebSocket Frame opCode=1 length=191
[firebase-websocket-worker] DEBUG io.netty.handler.codec.http.websocketx.WebSocket08FrameDecoder - Decoding WebSocket Frame opCode=1
[firebase-websocket-worker] DEBUG io.netty.handler.codec.http.websocketx.WebSocket08FrameDecoder - Decoding WebSocket Frame length=43
[firebase-websocket-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] WS message: {"t":"d","d":{"r":0,"b":{"s":"ok","d":""}}}
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Reset keepAlive. Remaining: 44945
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Handle new frame count: 1
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Parsed complete frame: {t=d, d={r=0, b={d=, s=ok}}}
[firebase-database-worker] DEBUG com.google.firebase.database.connection.Connection - [conn_0] Received data message: {r=0, b={d=, s=ok}}
[firebase-websocket-worker] DEBUG io.netty.handler.codec.http.websocketx.WebSocket08FrameDecoder - Decoding WebSocket Frame opCode=1
[firebase-websocket-worker] DEBUG io.netty.handler.codec.http.websocketx.WebSocket08FrameDecoder - Decoding WebSocket Frame length=75
[firebase-websocket-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] WS message: {"t":"d","d":{"r":1,"b":{"s":"ok","d":{"auth":null,"expires":1521576206}}}}
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Reset keepAlive. Remaining: 44957
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Handle new frame count: 1
[firebase-database-worker] DEBUG com.google.firebase.database.connection.WebsocketConnection - [ws_0] Parsed complete frame: {t=d, d={r=1, b={d={auth=null, expires=1521576206}, s=ok}}}
```

### CODE REVIEW

File: src/main/java/com/google/firebase/database/core/Repo.java [hunk 1]
```diff
@@ -67,14 +68,14 @@
 
   private static final String TRANSACTION_TOO_MANY_RETRIES = "maxretries";
   private static final String TRANSACTION_OVERRIDE_BY_SET = "overriddenBySet";
+
+  private static final Logger operationLogger = LoggerFactory.getLogger(Repo.class);
```
Comment (schmidt-sebastian) [inline src/main/java/com/google/firebase/database/core/Repo.java]: You are now using this in places were you previously used 'transactionLogger'. I'd suggest naming this just 'logger'.

File: src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java [hunk 2]
```diff
@@ -319,21 +313,18 @@ public void onDisconnect(Connection.DisconnectReason reason) {
 
   @Override
   public void onKill(String reason) {
-    if (logger.logsDebug()) {
+    if (logger.isDebugEnabled()) {
```
[the remark this answers is not in the document]
Comment (hiranya911) [reply src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java]: Done

File: src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java [hunk 3]
```diff
@@ -609,14 +595,10 @@ private void sendUnlisten(OutstandingListen listen) {
   }
 
   private OutstandingListen removeListen(ListenQuerySpec query) {
-    if (logger.logsDebug()) {
-      logger.debug("removing query " + query);
-    }
+    logger.debug("removing query {}", query);
```
[the remark this answers is not in the document]
Comment (hiranya911) [reply src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java]: Done

File: src/main/java/com/google/firebase/database/core/Repo.java (hunk 1 above)
Comment (hiranya911) [reply src/main/java/com/google/firebase/database/core/Repo.java]: Done

**Pull request merged**